### PR TITLE
GROMACS and LAMMPS craypat

### PR DESCRIPTION
**Pull request by victorusu**

### CODE REVIEW

**Comment by jgphpc** [comment]
No description provided.

**Comment by gppezzi** [comment]
@victorusu build passes but could you (shortly) report if they were tested or not?

**Comment by victorusu** [comment]
@jgphpc the crypt reports that I submitted on Madera were performed using these executables.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+# License::   MIT/GPL
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component of the policy:
+# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_2012-93.html
+
+name = 'GROMACS'
+version = "5.1.4"
+cudaversion = '8.0'
+craypatversion = '6.4.5'
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb]: Please use the version of the modulefile to be loaded as build_dependency.
E.g.:  patversion = ' 645-cuda'
Then you can use this variable in the build_dependency list.

File: easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles."""
+
+toolchain = {'name': 'CrayGNU', 'version': '2016.11'}
+toolchainopts = {'usempi': True, 'openmp': True, 'pic': True, 'verbose': False}
+
+source_urls = ['ftp://ftp.gromacs.org/pub/gromacs/']
+sources = [SOURCELOWER_TAR_GZ]
+
+builddependencies = [
+    ('perftools-cscs', '645-cuda', '', True),
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb]: Please ('perftools-cscs', '%s' %patversion, True)

File: easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-pat-6.4.5.eb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+#             George Tsouloupas <g.tsouloupas@cyi.ac.cy>, Kenneth Hoste
+# License::   MIT/GPL
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component of the policy:
+# http://hpcbios.readthedocs.org/en/latest/HPCBIOS_2012-93.html
+#
+name = 'GROMACS'
+version = "5.1.4"
+craypatversion = '6.4.5'
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-pat-6.4.5.eb]: Please use patversion = '645-nogpu'

File: easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-pat-6.4.5.eb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+sources = [SOURCELOWER_TAR_GZ]
+
+configopts = ' -DCMAKE_C_FLAGS="-g -fopenmp" -DCMAKE_CXX_FLAGS="-g -fopenmp" '
+
+skipsteps = ['test']
+
+# CMake and libxml2 dependencies with dummy toolchain
+builddependencies = [
+    ('CMake', '3.6.0'),
+    ('perftools-cscs','645-nogpu', '', True),
```
Comment (lucamar) [inline easybuild/easyconfigs/g/GROMACS/GROMACS-5.1.4-CrayGNU-2016.11-pat-6.4.5.eb]: Please use ('perftools-cscs','%s' %patversion, '', True)

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb [hunk 5]
```diff
@@ -0,0 +1,50 @@
+# Contributed by
+# Luca Marsella (CSCS)
+# Victor Holanda Rusu (CSCS)
+easyblock = 'MakeCp'
+
+name = 'LAMMPS'
+version = '30Jul16'
+release = 'r15407'
+cudaversion =  '8.0'
+craypatversion = '6.4.5'
```
Comment (lucamar) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb]: Please use patversion = '645-cuda'

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+# go to folder ./lib/gpu and create Makefile.gpu for sm_60 (the executable was failing on Pascal GPU without that)
+prebuildopts += ' pushd ../lib/gpu && sed -e "s/-march=bdver1//g" -e "s/sm_[0-9]*/sm_60/g" Makefile.xk7 > Makefile.gpu && '
+prebuildopts += ' make -f Makefile.gpu && popd && '
+# create Makefile.omp and correct Makefile.mpi
+prebuildopts += ' sed -e \'s/mpicxx/CC -fopenmp/\' ./MAKE/Makefile.mpi > ./MAKE/Makefile.omp && '
+prebuildopts += ' sed -i -e \'s/mpicxx/CC/\' ./MAKE/Makefile.mpi && '
+buildopts = [ ' mpi ', ' omp ' ]
+
+builddependencies = [
+    ('perftools-cscs', '645-cuda', '', True),
```
Comment (lucamar) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-cuda-8.0-pat-6.4.5.eb]: Please use ('perftools-cscs', '%s' %patversion, '', True)

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-pat-6.4.5.eb [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Contributed by
+# Luca Marsella (CSCS)
+# Victor Holanda Rusu (CSCS)
+easyblock = 'MakeCp'
+
+name = 'LAMMPS'
+version = "30Jul16"
+release = 'r15407'
+craypatversion = '6.4.5'
```
Comment (lucamar) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-pat-6.4.5.eb]: Please use patversion = '645-nogpu'

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-pat-6.4.5.eb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+prebuildopts += ' make yes-standard yes-user-reaxc yes-user-omp yes-kspace yes-misc yes-molecule yes-rigid yes-user-misc && '
+prebuildopts += ' make no-voronoi no-poems no-meam no-kim no-gpu && '
+prebuildopts += ' make package-update && '
+prebuildopts += ' pushd ../lib/reax && make -f Makefile.gfortran && popd && '
+prebuildopts += ' sed -e \'s/mpicxx/CC -fopenmp/\' ./MAKE/Makefile.mpi > ./MAKE/Makefile.omp && '
+prebuildopts += ' sed -i -e \'s/mpicxx/CC/\' ./MAKE/Makefile.mpi && '
+buildopts = [ ' mpi ', ' omp ' ]
+
+builddependencies = [
+    ('perftools-cscs', '645-nogpu', '', True),
```
Comment (lucamar) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-30Jul16-CrayGNU-2016.11-pat-6.4.5.eb]: Please use ('perftools-cscs', '%s' %patversion, '', True)

**Pull request merged**